### PR TITLE
Date class and other project initial project setups

### PR DESCRIPTION
**Pull request by Seif-Mamdouh**

### CODE REVIEW

File: Event.java
Comment (Seif-Mamdouh) [inline Event.java:1]: you can ignore that, for idk why I commented it lol

File: eventOrganizer/Date.java [hunk 1]
```diff
@@ -1,4 +1,74 @@
...
+        int [] daysInMonth = {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+
+        if(isLeapYear(year)){
+            daysInMonth[2] = 29;
+        };
+
+        return day >= 1 && day <= daysInMonth[month];
+
+    }
+    public int compareTo(Date otherDate) {
```
Comment (Seif-Mamdouh) [inline eventOrganizer/Date.java:34]: Let me know if I did this correctly

File: eventOrganizer/Date.java [hunk 2]
```diff
@@ -1,4 +1,74 @@
...
+    private int year;
+    private int month;
+    private int day;
+
+    public Date(int year, int month, int day){
+        this.year = year;
+        this.month = month;
+        this.day = day;
+    }
+
```
Comment (Seif-Mamdouh) [inline eventOrganizer/Date.java:18]: According the project docs. The date class has isValid() method that checks if the date is a leap year or not

File: eventOrganizer/Date.java [hunk 3]
```diff
@@ -1,4 +1,74 @@
...
+    public boolean isValid(){
+
+        if (month < 1 || month > 12){
+            return false;
+        }
+
+        int [] daysInMonth = {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
```
Comment (Seif-Mamdouh) [inline eventOrganizer/Date.java:25]: I'm not sure if this was the best way to implement the numbers of the months

File: eventOrganizer/Date.java [hunk 4]
```diff
@@ -1,4 +1,74 @@
...
+    }
+
+    private boolean isLeapYear(int year){
+        if((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
+            return true;
+        }
+        return false;
+    };
+
+    public static void main(String[] args) {
```
Comment (Seif-Mamdouh) [inline eventOrganizer/Date.java:53]: This is just to test my code and just incase you want to switch to my branch and test. That's why I left the print statements there

File: Event.java [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
-    private String location;
-    private TimeSlot time;
-    private String Contact;
-
-public static void main(String [] args){
-    new
-}
-
-}
+//public class Event {
```
Comment (mmm606) [inline Event.java:1]: Probably shouldn't make a change to every class in this pr sort of out of the scope. Why not just make changes to Date? Most of the changes in classes other than Date are just comments we don't really need.

**Review by mmm606 — APPROVED** [review]
Approved, but next time just make the changes needed. No need for random comments.

Also not sure if you are super serious about these pr reviews but if you are, give a summary for each pr, it is good practice.

File: eventOrganizer/Date.java
Comment (mmm606) [inline eventOrganizer/Date.java:1]: Add the javadocs and everything as per the standard prof is forcing

File: Event.java (hunk 5 above)
Comment (Seif-Mamdouh) [reply Event.java:1]: Roger that

File: eventOrganizer/Date.java
Comment (Seif-Mamdouh) [reply eventOrganizer/Date.java:1]: okay will do

**Pull request merged**